eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Rules/Chain.hs

```diff
@@ -340,8 +336,7 @@ chainTransition =
             chainChecksData = pparamsToChainChecksPParams pp
             bhView = makeHeaderView bh
 
-        maxpv <- liftSTS $ asks maxMajorPV
-        case chainChecks maxpv chainChecksData bhView of
+        case chainChecks chainChecksData bhView of
```

Comment:
```suggestion
        case chainChecks (succ (eraProtVerHigh @era)) chainChecksData bhView of
```

Reply:
@lehins, it seems like `succ (eraProtVerHigh @era)` evaluates to `3` because the `@era` is `Shelley` and this causes test failures. `pvMajor $ pp ^. ppProtocolVersionL` fixes it but I don't know if this discrepancy between `succ (eraProtVerHigh @era)` and `pvMajor $ pp^. ppProtocolVersionL` is okay / expected, so I thought I would point it out.

Reply:
In the matter of fact looking closer at the meaning of the `chainChecks` function, makes me realize it should be `eraProtVerHigh @era`. 

The reason why tests started failing is because generators for the tests are incorrect. With this change you'd get errors like this:

```
ObsoleteNodeCHAIN (Version 9) (Version 2)
```

Which means generators are trying to use protocol version `9` in Shelley, which makes no sense whatsoever.

I am gonna try and fix generators and see if we can have it done properly.

The reason why `pvMajor $ pp ^. ppProtocolVersionL` works, is because then `chainChecks` always compares protocol version to itself, which is sort of useless

Reply:
@lehins ah, I see. I was afraid that this will be the case, that we are asserting wrong behaviour. In the meantime, I'll go back to my other issues but if your hands are full, I'm happy to look at those generators and have a go at fixing this properly.

Reply:
> I'm happy to look at those generators and have a go at fixing this properly.

No worries, I had it half way done last week, just didn't have the time to finish it. Here is the PR that you can use to get this one unblocked: https://github.com/IntersectMBO/cardano-ledger/pull/4576